src/main/kotlin/com/openlattice/chronicle/controllers/SurveyController.kt

```diff
@@ -0,0 +1,185 @@
+import com.openlattice.chronicle.survey.SurveyApi.Companion.QUESTIONNAIRE_ID_PATH
+import com.openlattice.chronicle.survey.SurveyApi.Companion.QUESTIONNAIRE_PATH
+import com.openlattice.chronicle.survey.SurveyApi.Companion.START_DATE
+import com.openlattice.chronicle.survey.SurveyApi.Companion.STUDY_ID
+import com.openlattice.chronicle.survey.SurveyApi.Companion.STUDY_ID_PATH
+import com.openlattice.chronicle.survey.SurveyApi.Companion.TYPE
+import com.openlattice.chronicle.util.ChronicleServerUtil
+import org.springframework.format.annotation.DateTimeFormat
+import org.springframework.http.MediaType
+import org.springframework.web.bind.annotation.*
+import java.time.LocalDate
+import java.time.OffsetDateTime
+import java.time.format.DateTimeFormatter
+import java.util.*
+import javax.inject.Inject
+import javax.servlet.http.HttpServletResponse
+
+/**
+ * @author alfoncenzioka &lt;alfonce@openlattice.com&gt;
+ */
+
+@RestController
+@RequestMapping(CONTROLLER)
+class SurveyController @Inject constructor(
+    val surveysService: SurveysService,
+    val downloadService: DataDownloadService,
+    val idGenerationService: HazelcastIdGenerationService,
+    override val authorizationManager: AuthorizationManager,
+    override val auditingManager: AuditingManager
+) : SurveyApi, AuthorizingComponent {
+
+    @Timed
+    @GetMapping(
+        path = [STUDY_ID_PATH + PARTICIPANT_PATH + PARTICIPANT_ID_PATH + APP_USAGE_PATH],
+        produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    override fun getAppUsageSurveyData(
+        @PathVariable(STUDY_ID) studyId: UUID,
+        @PathVariable(PARTICIPANT_ID) participantId: String,
+        @RequestParam(value = START_DATE) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) startDateTime: OffsetDateTime,
+        @RequestParam(value = END_DATE) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) endDateTime: OffsetDateTime
+    ): List<AppUsage> {
+        return surveysService.getAppUsageData(studyId, participantId, startDateTime, endDateTime)
+    }
+
+    @Timed
+    @PostMapping(
+        path = [STUDY_ID_PATH + PARTICIPANT_PATH + PARTICIPANT_ID_PATH + APP_USAGE_PATH],
+        consumes = [MediaType.APPLICATION_JSON_VALUE],
+    )
+    override fun submitAppUsageSurvey(
+        @PathVariable(STUDY_ID) studyId: UUID,
+        @PathVariable(PARTICIPANT_ID) participantId: String,
+        @RequestBody surveyResponses: List<AppUsage>
+    ) {
+        surveysService.submitAppUsageSurvey(studyId, participantId, surveyResponses)
+    }
+
+    @Timed
+    @PostMapping(
+        path = [STUDY_ID_PATH + QUESTIONNAIRE_PATH],
+        consumes = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    override fun createQuestionnaire(
+        @PathVariable(STUDY_ID) studyId: UUID,
+        @RequestBody questionnaire: Questionnaire
+    ): UUID {
+        ensureWriteAccess(AclKey(studyId))
+        val id = idGenerationService.getNextId()
```

Comment:
I think it makes more sense to have id generation happen in the service, i.e. the lower in the stack the better.

Reply:
Yeah, it's worth pushing down auditing to the service as well.